### DISCUSSION TITLE
How to sum a currency format?

### DISCUSSION THREAD

**Discussion by ivanfa5** [Power-Components/livewire-powergrid#847] [Q&A] [answered]
Hi, I am totally confused about how to sum from a field that contains currency closure.

I already follow some of these documentaries:
1. https://livewire-powergrid.com/table/add-columns.html#currency
2. https://livewire-powergrid.com/table/column-summary.html#withsum
3. https://v2.livewire-powergrid.com/#/table/column-summary?id=column-summary
4. https://v2.livewire-powergrid.com/#/table/add-columns?id=closure-examples

Also, I searched this issue in issues and discovered that it was a bug, and it has been closed. 
https://github.com/Power-Components/livewire-powergrid/issues/281

But none of those work for me.
My goal is the same as the image shown in the documentary.
![image](https://user-images.githubusercontent.com/77907893/218105291-3ef2d7e0-19b7-4dc5-abb5-6205c7743faf.png)
The field contains a currency with a sum on the header.

I also try it in powergrid demo, the problem still encounter
![image](https://user-images.githubusercontent.com/77907893/218105872-9e9d862f-b313-41c8-8354-22e388c003c5.png)
The image shown above is of me trying to:
1. Re-format calories field values to a currency, the currency work, but the sum not.
2. Sum 'SALES PRICE' field that has been in a currency by default with step from link number 2 & 3, It doesn't work.

I tried this for calories
`->addColumn('calories', function (Dish $dish) {
            return 'R$ ' . number_format(e($dish->calories), 2, ',', '.');
            })`
and this
`->addColumn('calories', function (Dish $dish) {
            $fmt = new NumberFormatter('pt_PT', NumberFormatter::CURRENCY);
            return $fmt->formatCurrency($dish->calories, "EUR");
            })`
with this in column
`Column::add()
                ->title(__('Calories'))
                ->field('calories')
                ->makeInputRange('calories')
                ->sortable()
                ->withSum('Sum', true, false),`
still not working
![image](https://user-images.githubusercontent.com/77907893/218107152-69831a9b-7582-4322-babd-5e8e95a5223a.png)

or let's try to add currency from the "PRICE" column that has successfully shown the sum but doesn't have currency by default.
`->addColumn('price', function (Dish $dish) {
                $fmat = new NumberFormatter('pt_PT', NumberFormatter::CURRENCY);
                return $fmat->formatCurrency($dish->price, "EUR");
                })`
The sum is gone
![image](https://user-images.githubusercontent.com/77907893/218108859-426d8a2b-b8c4-4042-a937-226369a44645.png)
 

**Could someone please show me or provide the full code for adding currencies with sum?** 
The documentary didn't help me, and it only showed a piece of code, and the demo also doesn't come with an example like the image shown in links 3 and 4 above. I've been searching for a solution for a couple weeks and have found nothing.
I'm begging for help.
Thanks in advance

**Comment by dansysanalyst** [comment]
Hi @ivanfa5,


Assuming you have these two columns, `price` as the raw amount and `price_BRL` as Brazilian Real currency format.


```php 

    public function addColumns(): PowerGridEloquent
    {
        return PowerGrid::eloquent()
            ->addColumn('id')
            
            //1000.00
            ->addColumn('price')
            
            //R$ 1.000,00
            ->addColumn('price_BRL', fn (Dish $dish) =>'R$ ' . number_format(e($dish->price), 2, ',', '.'));
    }
```

You can then do:

```php
 public function columns(): array
    {
        return [
            Column::add()
                ->title(__('ID'))
                ->field('id', 'dishes.id')
                ->searchable()
                ->sortable(),

            Column::add()
                ->title(__('Price'))
                ->field( 'price_BRL', 'price')
                ->withSum('Total em R$', true, true),

        ];
    }
```

![CleanShot 2023-02-10 at 16 12 27@2x](https://user-images.githubusercontent.com/79267265/218126791-609ec616-7dbc-40b2-87a1-14fcef428416.png)

I will update the doc to make it more clear 😊

**Reply by ivanfa5** [reply]
Hi @dansysanalyst,

Thank you for your fast response.
Your solution help me and give me more understanding about formatting.
I tried it and it worked, but it leads me to another question. 

**How do I move the summary into the `price` field without making a new blank field next to it?**
![image](https://user-images.githubusercontent.com/77907893/218136090-8c68de95-c0bd-4218-a12e-d350dfa73545.png)

I have tried using your code with removing either `'price_BRL'` or `'price'` also tried to remove `->addColumn('price')` none of those work. The output either shows that the sum gone or it remains in the next column header. 
![image](https://user-images.githubusercontent.com/77907893/218142468-9e5d3a47-740a-4799-a1ee-a07a81b5880b.png)
![image](https://user-images.githubusercontent.com/77907893/218142670-d722d97c-98fe-4dbb-aafa-256bb80264dd.png)
![image](https://user-images.githubusercontent.com/77907893/218142717-adefcefc-1dfd-403b-bdc9-9742bfc5ecb5.png)

I also tried to use Column::make instead Column::add, hoping that it'll be a single column. still not working.

What I expected is literally the same as in the documentary.
![image](https://user-images.githubusercontent.com/77907893/218143780-1ae82573-3600-4f02-b194-5b1822a47f99.png)
(https://livewire-powergrid.com/table/column-summary.html#withsum)

looking forward for the solution.
Thanks

**Comment by dansysanalyst** [comment, marked as answer]
You can create a [custom theme](https://livewire-powergrid.com/table/custom-theme.html) and change the CSS.

For instance:

```php
//...
class MyTheme extends Tailwind
{
    public string $name = 'tailwind';

    public function table(): Table
    {
        return Theme::table('rounded-lg min-w-full border border-slate-200 dark:bg-slate-600 dark:border-slate-500')
             //...
            ->tdBodyTotalColumns('px-3 py-2 whitespace-nowrap dark:text-slate-200 text-sm text-slate-600 text-left space-y-2');   //Align left
//...
```
